pest/src/parser_state.rs

```diff
@@ -896,6 +896,33 @@ impl<'i, R: RuleType> ParserState<'i, R> {
         self.match_string(string)
     }
 
+    /// Matches part of the state of the stack.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// TODO
+    /// ```
+    #[inline]
+    pub fn stack_slice_peek(mut self: Box<Self>, start: i32, end: Option<i32>) -> ParseResult<Box<Self>> {        
```

Comment:
The implementation is now there, now I only need to get it to run.

I wrote no tests yet, but since I replaced `stack_match_peek`’s body with a call to this, its test failing shows that this doesn’t work yet. Maybe someone sees the error?

Reply:
That’s the question, I need to step through it using a debugger.